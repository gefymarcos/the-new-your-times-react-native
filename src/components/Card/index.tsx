import * as React from 'react'
import { Text, View, StyleSheet } from 'react-native'
import theme from '../../styles/theme'

type cardProps = {
  title: string,
  paragraph: string,
  url: string
}

const Card = ({ title, paragraph, url }: cardProps) => {
  return (
    <View style={styles.card}>
      <Text style={styles.title}>{title}</Text>
      <Text style={styles.paragraph}>
        {paragraph}
      </Text>
      <Text style={styles.url}>
        {url}
      </Text>
    </View>
  )
}

const styles = StyleSheet.create({
  card: {
    margin: 8,
    paddingBottom: 16,
    justifyContent: 'center',
    borderBottomColor: theme.colors.black,
    borderBottomWidth: 1,
  },
  url: {
    marginHorizontal: 16,
    fontSize: 12,
    color: theme.colors.lightGray
  },
  title: {
    fontWeight: 'bold',
    fontSize: 18,
    marginVertical: 8,
    marginHorizontal: 16,
    color: theme.colors.black
  },
  paragraph: {
    fontSize: 16,
    lineHeight: 24,
    marginVertical: 8,
    marginHorizontal: 16,
    color: theme.colors.black
  },
})

export default Card
