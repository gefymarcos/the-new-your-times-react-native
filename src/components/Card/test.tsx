import React from 'react'
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'
import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'

import { fireEvent, render } from '@testing-library/react-native'

import Card from '.'

jest.useFakeTimers()

const props = {
  title: 'Title of card',
  abstract: 'Description',
  url: 'https://myurl.com.br'
}

describe('<Card />', () => {
  const middlewares = [thunk]
  const mockStore = configureStore(middlewares)

  it('should render correctly', () => {
    const { getAllByText } = render(
      <Provider store={mockStore({})}>
        <Card {...props} />
      </Provider>
    )

    const title = getAllByText(/Title of card/i)
    expect(title).toBeDefined()
  })

  it('should render correctly', () => {
    const store = mockStore({})
    const { getAllByText } = render(
      <Provider store={store}>
        <Card {...props} />
      </Provider>
    )

    const title = getAllByText(/Title of card/i)
    expect(title).toBeDefined()
  })

  it('should dispatched action with press a card', () => {
    const store = mockStore({})
    const { getAllByText } = render(
      <Provider store={store}>
        <Card {...props} />
      </Provider>
    )

    const title = getAllByText(/Title of card/i)
    expect(title).toBeDefined()
    fireEvent(title[0], 'press')
    const actions = store.getActions()
    expect(actions.length).toBe(1)
    expect(actions[0].type).toEqual('ARTICLE_DETAIL_FULLFILLED')
  })

  it('should render correctly based on snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={mockStore({})}>
          <Card {...props} />
        </Provider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
