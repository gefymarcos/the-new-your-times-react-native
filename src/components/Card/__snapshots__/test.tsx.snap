// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should render correctly based on snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderBottomColor": "#212121",
      "borderBottomWidth": 1,
      "marginHorizontal": 16,
    }
  }
>
  <View
    style={
      Object {
        "justifyContent": "center",
        "paddingVertical": 32,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#212121",
          "fontSize": 18,
          "fontWeight": "600",
          "marginHorizontal": 16,
          "marginVertical": 8,
        }
      }
    >
      Title of card
    </Text>
  </View>
</View>
`
